numpy/lib/format.py

```diff
@@ -266,21 +371,12 @@ def header_data_from_array_1_0(array):
         This has the appropriate entries for writing its string representation
         to the header of the file.
     """
-    d = {'shape': array.shape}
-    if array.flags.c_contiguous:
-        d['fortran_order'] = False
-    elif array.flags.f_contiguous:
-        d['fortran_order'] = True
-    else:
-        # Totally non-contiguous data. We will have to make it C-contiguous
-        # before writing. Note that we need to test for C_CONTIGUOUS first
-        # because a 1-D array is both C_CONTIGUOUS and F_CONTIGUOUS.
-        d['fortran_order'] = False
```

Comment:
I have the feeling that the part about "[w]e will have to make it C-contiguous before writing" belongs somewhere else. At least I do not see the connection between setting `d['fortran_order'] = False` and making it C-contiguous before writing. The part that c_contiguous implies not f_contiguous could be explained above the refactored code...

Reply:
Just to fill you in: The `c_contiguous` and `f_contiguous` flags have somewhat confusing meanings which it is too late for us to change for back-compat reasons. It is possible for arrays to be both c-contiguous and f-contiguous (eg, all non-unusually-strided 1d arrays are like this). It is also possible to be neither (eg, for weird strides). I think the logic here is that if it is neither, we will later copy it into a C-contiguous buffer, so `fortran_ordered` should be False.

I agree something about this code seems a bit messy, since we repeat the check for f-order in two places: Here, and in `write_array` (which is where we copy to a C-contiguous buffer if needed). But maybe it's OK...